File: src/lib/helpers/tauriCommands/appSettingsCommands.ts
import {invoke} from "@tauri-apps/api/core";
import type {AppSettings} from "$lib/models/Settings";

// Save the settings in the configuration file
async function saveSettings(
    theme: string,
    language: string,
    installPath: string | undefined,
) {
    try {
        await invoke("save_settings_command", {
            settings: {
                theme,
                language,
                installPath,
            }
        });
    }
    catch (e) {
        console.error(e);
        throw new Error('Failed to save settings');
    }
}

async function getSettings() {
    try {
        const settings: AppSettings = await invoke<AppSettings>("read_settings_command");
        if (settings){
            return settings;
        }
        else {
            throw new Error('Failed to read settings');
        }
    } catch (error) {
        console.error('Error reading settings', error);
        throw error;
    }
}

export {
    saveSettings,
    getSettings
};

File: src-tauri/src/commands/app_settings_commands.rs
// Documentation available at:
// https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/store#usage-from-rust
// plugin is in beta, so the API may change

use tauri::{AppHandle, Manager};

use crate::models::app_settings::AppSettings;
use crate::services::settings_service::{read_settings, save_settings};

#[tauri::command]
pub async fn read_settings_command(app: AppHandle) -> Result<AppSettings, String> {
   match read_settings(app) {
       Ok(settings) => {
          Ok(settings)
       }
       Err(error) => {
            Err(error)
       }
   }
}

#[tauri::command]
pub async fn save_settings_command(app: AppHandle, settings: AppSettings) -> Result<(), String> {
    match save_settings(app, settings).await {
        Ok(_) => {
            Ok(())
        }
        Err(error) => {
            Err(error)
        }
    }
}
